src/detail/CommentInputBox.tsx

```diff
@@ -0,0 +1,107 @@
+import axios from "axios";
+import Image from "next/image";
+import React, { useState } from "react";
+import { CommentWithReComments } from "./CommentBox";
+import CommentInput from "./CommentInput";
+
+interface CommentInputBoxProps {
+  type: "re" | "com";
+  postId: number;
+  commentId?: number;
+  onFetchComments: (newComments: CommentWithReComments[]) => void;
+}
+
+const CommentInputBox = ({
+  type,
+  postId,
+  commentId,
+  onFetchComments,
+}: CommentInputBoxProps) => {
+  const defaultCommentState = {
+    content: "",
+    isHidden: false,
+  };
+
+  const [comment, setComment] = useState(defaultCommentState);
+
+  const onChangeComment = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setComment((prevComment) => ({
+      ...prevComment,
+      content: e.target.value,
+    }));
+  };
+
+  const onChangeHidden = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setComment({ ...comment, isHidden: e.target.checked });
+  };
+
+  const onClickSubmit = async () => {
+    // TODO: 서버에 요청 추가
+    if (type === "com") {
+      // 댓글
+      try {
+        const data = { postId, ...comment };
```

Comment:
이부분만 빼고 위아래 코드가 다 동일한데 조건문을 조금 수정하면 하나로 사용할 수 있을것같습니다!
```suggestion
        
        const data = {
          postId,
          ...comment,
          ...(type === "com" ? { commentId } : {}),
        };
```